storage/mysql/tree_storage.go

```diff
@@ -82,6 +87,21 @@ func OpenDB(dbURL string) (*sql.DB, error) {
 		return nil, err
 	}
 
+	// Limit the number of open connections to the database, and periodically
+	// report the currently-open connections as a Prometheus gauge.
+	db.SetMaxOpenConns(*maxConns)
+	openConnGauge := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "db_open_connections",
+			Help: "Currently open connections to the database",
+		})
+
+	go func() {
```

Comment:
There's nothing to stop this goroutine, so if anything does repeated `NewStorageProvider()` calls then these goroutines will stack up.  I'm not sure if anything actually does that (maybe tests?), but it would be neater to stop this gauge-updater.  So maybe move it up into `mysqlProvider` and have `mysqlProvider` close a channel to terminate the gauge-updater?